src/main/java/com/gmail/goosius/siegewar/listeners/SiegeWarTownEventListener.java

```diff
@@ -268,22 +271,39 @@ public void onTownRankGivenToPlayer(TownAddResidentRankEvent event) {
 
 	/**
 	 * If this is a peaceful town, UN-CANCEL the town spawn event
-	 * 
+	 * If the town's nation has an active offensive siege and a battle session is active, CANCEL town spawn event to outposts
+	 *
 	 * Lowest priority so that more important features like the siegezone-tp-block can can take precedence 
 	 *
 	 * @param event town spawn event
 	 */
 	@EventHandler(priority = EventPriority.LOWEST)
 	public void on(TownSpawnEvent event) {
-		if(SiegeWarSettings.getWarSiegeEnabled()
-				&& SiegeWarSettings.getWarCommonPeacefulTownsEnabled()
+		if(SiegeWarSettings.getWarSiegeEnabled()) {
+			Town toTown = event.getToTown();
+
+			if (SiegeWarSettings.getWarCommonPeacefulTownsEnabled()
 				&& SiegeWarSettings.isPeacefulTownPublicSpawnEnabled()) {
-			if (event.isCancelled()
-					&& SiegeWarTownPeacefulnessUtil.isTownPeaceful(event.getToTown())
-					&& event.getToTown().isPublic()) {
-				event.setCancelled(false); //UN-Cancel the event
+				if (event.isCancelled()
+						&& SiegeWarTownPeacefulnessUtil.isTownPeaceful(toTown)
+						&& toTown.isPublic()) {
+					event.setCancelled(false); //UN-Cancel the event
+				}
+			}
+
+			try {
+				//If enabled in config, prevent teleportation to outposts during offensive siege & battle session
+				if (!SiegeController.getActiveOffensiveSieges(toTown.getNation()).isEmpty()
+						&& toTown.getTownBlock(WorldCoord.parseWorldCoord(event.getTo())).isOutpost()
+						&& SiegeWarSettings.getWarSiegeOutpostTeleportationDisabled() && BattleSession.getBattleSession().isActive()) {
+					Translator translator = Translator.locale(event.getPlayer());
+
+					event.setCancelMessage(translator.of("siegewar_plugin_prefix") + translator.of("msg_err_cannot_spawn_outpost_during_battle_session"));
+					event.setCancelled(true); //Stop the teleport
+				}
+			} catch (TownyException e) {
+				e.printStackTrace();

```

Comment:
```suggestion
			Nation nation = toTown.getNationOrNull();
			TownBlock tb = WorldCoord.parseWorldCoord(event.getTo()).getTownBlockOrNull();
			if (nation == null || tb == null)
				return;
			//If enabled in config, prevent teleportation to outposts during offensive siege while battle session is active.
			if (!SiegeController.getActiveOffensiveSieges(nation).isEmpty()
					&& tb.isOutpost()
					&& SiegeWarSettings.getWarSiegeOutpostTeleportationDisabled() && BattleSession.getBattleSession().isActive()) {
				Translator translator = Translator.locale(event.getPlayer());

				event.setCancelMessage(translator.of("siegewar_plugin_prefix") + translator.of("msg_err_cannot_spawn_outpost_during_battle_session"));
				event.setCancelled(true); //Stop the teleport

```

This is safer to use, the code you supplied would throw an error in the log when the town has no nation.